### PR TITLE
001 - Create new Delivery employee

### PR DESCRIPTION
**Pull request by mateenparkar**

### CODE REVIEW

File: src/main/java/org/kainos/ea/api/DeliveryService.java [hunk 1]
```diff
@@ -1,2 +1,42 @@
...
+import org.kainos.ea.cli.DeliveryEmployee;
+import org.kainos.ea.cli.DeliveryRequest;
+import org.kainos.ea.client.DeliveryEmployeeDoesNotExistException;
+import org.kainos.ea.client.FailedToCreateDeliveryEmployeeeException;
+import org.kainos.ea.db.DeliveryDao;
+
+import java.sql.SQLException;
+
+public class DeliveryService {
+    private DeliveryDao deliveryDao = new DeliveryDao();
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/api/DeliveryService.java]: Can this be improved to follow the D in solid. And inject the DAO through the constructor

File: src/main/java/org/kainos/ea/cli/DeliveryRequest.java [hunk 2]
```diff
@@ -1,2 +1,55 @@
...
+
+    public void setNI(String NI) {
+        this.NI = NI;
+    }
+
+    String bankNumber;
+    String NI;
+
+    @JsonCreator
+    public DeliveryRequest(@JsonProperty("name") String name, @JsonProperty("salary") double salary, @JsonProperty("bankNumber") String bankNumber, @JsonProperty("NI") String NI) {
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/cli/DeliveryRequest.java:48]: NI should be ni if following the same naming convension for all the parameters

File: src/main/java/org/kainos/ea/db/DeliveryDao.java [hunk 3]
```diff
@@ -1,2 +1,68 @@
-package org.kainos.ea.db;public class DeliveryDao {
+package org.kainos.ea.db;
+
+import org.kainos.ea.cli.DeliveryEmployee;
+import org.kainos.ea.cli.DeliveryRequest;
+
+import java.sql.*;
+
+public class DeliveryDao {
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/db/DeliveryDao.java:10]: Does the class name match the entity this DAO is for

File: src/main/java/org/kainos/ea/db/DeliveryDao.java [hunk 4]
```diff
@@ -1,2 +1,68 @@
...
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/db/DeliveryDao.java]: Can you inject the connector via the constructor

File: src/main/java/org/kainos/ea/api/DeliveryService.java [hunk 5]
```diff
@@ -1,2 +1,42 @@
...
+            int id = deliveryDao.createDeliveryEmployee(deliveryRequest);
+
+            if (id == -1) {
+                throw new FailedToCreateDeliveryEmployeeeException();
+            }
+            return id;
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+        }
+        return -1;
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/api/DeliveryService.java]: If we fail to create due to a SQL error currently we would log a message and just return -1 to the controller which would return a 200 response

**Review by TomBillingtonUK — APPROVED** [review]

**Review by CallumE74 — APPROVED** [review]

**Pull request merged**